### PR TITLE
Promise branch

### PR DESCRIPTION
**Pull request by promisepreston**
#### What does this PR do?
This PR upgrades the Ruby and Rails versions and also adds webpacker.

#### Description of Task to be completed?

1. Upgrading the rails version from 6.0.2 to 6.1.0
2. Upgrading the ruby version from 2.5.3 to 2.7.2
3. Adding support for .env file for the database credentials
4. Organizing the `Gemfile`
5. Adding webpacker and procfiles for on-the-fly assets-precompilation.
6. Upgrading some gems in the Gemfile as well.
7. Removing the turbolinks gem from the application

#### How should this be manually tested?
You can check the `Gemfile` for the upgrades done.
You can also check the `config/database.yml` file for the modified database environment variables
You will need to create a `.env `file to use in your development environment that will contain your database environment values.

#### Any background context you want to provide?

More background context will be provided in the updated `README.md` file that will be added soon.

### CODE REVIEW

File: config/database.yml [hunk 1]
```diff
@@ -17,21 +17,23 @@
...
 development:
   <<: *default
-  database: openemrs_development
 
   # The specified database role being used to connect to postgres.
   # To create additional roles in postgres see `$ createuser --help`.
   # When left blank, postgres will use the default role. This is
   # the same name as the operating system user that initialized the database.
-  #username: openemrs
+  #username: mailing_list
```
Comment (robocopkaka) [inline config/database.yml]: Can we stick with `openemrs` here

**Comment by robocopkaka** [comment]
Could you rename the branch for this PR so it shows what it's trying to achieve

File: config/database.yml (hunk 1 above)
Comment (promisepreston) [reply config/database.yml]: Oh that's a mistake from me

**Comment by promisepreston** [comment]
> Could you rename the branch for this PR so it shows what it's trying to achieve

Oh okay. If you feel so. Most times I just want to have one branch that I make commits. Then the commits and the PR can explain what I'm trying to achieve.

**Comment by robocopkaka** [comment]
I get. 

> > Could you rename the branch for this PR so it shows what it's trying to achieve
> 
> Oh okay. If you feel so. Most times I just want to have one branch that I make commits. Then the commits and the PR can explain what I'm trying to achieve.

I understand but feel its better that branch names also depict their intended goals

**Comment by promisepreston** [comment]
> I get.
> 
> > > Could you rename the branch for this PR so it shows what it's trying to achieve
> > 
> > 
> > Oh okay. If you feel so. Most times I just want to have one branch that I make commits. Then the commits and the PR can explain what I'm trying to achieve.
> 
> I understand but feel its better that branch names also depict their intended goals

Alright then, I will make the changes to it and push again now.

**Comment by promisepreston** [comment]
I will close this PR and create another one then

**Pull request closed without merging**